docs/HANDLED_RULES.md

```diff
@@ -203,11 +203,11 @@ Example:
 
 #### Synchronization should not be based on Strings or boxed primitives ([Sonar Rule 1860](https://rules.sonarsource.com/java/RSPEC-1860))
 
-Objects which are pooled and potentially reused should not be used for synchronization. If they are, it can cause unrelated threads to deadlock with unhelpful stacktraces. Specifically, String literals, and boxed primitives such as Integers should not be used as lock objects because they are pooled and reused.
+Objects which are pooled, such as Strings or boxed primitives and potentially reused should not be used for synchronization, since that can cause deadlocks. The transformation will do the following. If the lock is a field of the current class where the synchronization block is in, then it will simply adding a new field as an Object lock. If the lock is obtained from another object through the get method, it will add a new field for the new object lock and a new method return the object. 
```

Comment:
- "such as Strings or boxed primitives" -> "such as Strings or boxed primitives,"
- "since that can" -> "since they can"
- "it will simply adding" -> "it will simply add"
- "Object" -> "`Object`"
- "get method" -> "`get` method"
- "new method return the object." -> "new method to get the object."